tests/lax_numpy_indexing_test.py

```diff
@@ -871,6 +871,15 @@ def testTrivialGatherIsntGenerated(self):
     jaxpr = jax.make_jaxpr(lambda x: x[:4])(np.arange(4))
     self.assertEqual(len(jaxpr.jaxpr.eqns), 0)
 
+  @parameterized.named_parameters(
+      {"testcase_name": f"_{idx}", "idx": idx}
+      for idx in (-3, 0, 5, np.array(6), jnp.array(7)))
+  def testConstantIndexing(self, idx):
+    x = jnp.arange(10)
+    jaxpr = jax.make_jaxpr(lambda: x[idx])()
+    self.assertEqual(len(jaxpr.jaxpr.eqns), 1)
+    self.assertEqual(jaxpr.jaxpr.eqns[0].primitive, lax.gather_p)
```

Comment:
I might add another test here to confirm that if the index is a static array with `ndim > 0`, that we use traced operations rather than treating the index array as a constant. From your code I don't think there's any chance that would slip through, but it would be useful if there are any future refactorings.

Reply:
I think that's already covered in the following test, isn't it?
https://github.com/google/jax/blob/0ed29b63f08fa404b1e837b4ab8b96b481c4b78f/tests/lax_numpy_indexing_test.py#L623-L634

Reply:
Ah, I missed that, thanks